src/main/drivers/serial.c

```diff
@@ -37,13 +37,16 @@ uint32_t serialGetBaudRate(serialPort_t *instance)
 
 void serialWrite(serialPort_t *instance, uint8_t ch)
 {
-    instance->vTable->serialWrite(instance, ch);
+    while (!serialTxBytesFree(instance)) {
+    };
+
+	instance->vTable->serialWrite(instance, ch);
 }
```

Comment:
Definitely. There's already a circular buffer in vcp4 at the lower layer, so I'd just consider implementing it one layer down rather than in serial_usb_vcp.